lib/chef/resource/windows_security_policy.rb

```diff
@@ -83,6 +84,58 @@ class WindowsSecurityPolicy < Chef::Resource
       description: "Policy value to be set for policy name."
 
       load_current_value do |desired|
+        current_state = load_security_options
+
+        if desired.secoption == "ResetLockoutCount"
+          if desired.secvalue.to_i > 30
+            raise Chef::Exceptions::ValidationFailed, "The \"ResetLockoutCount\" value cannot be greater than 30 minutes"
+          end
+        end
+        if desired.secoption == "ResetLockoutCount" || desired.secoption == "LockoutDuration"
+          if current_state["LockoutBadCount"] == "0"
+            raise Chef::Exceptions::ValidationFailed, "#{desired.secoption} cannot be set unless the \"LockoutBadCount\" security policy has been set to a non-zero value"
+          else
+            secvalue current_state[desired.secoption.to_s]
+          end
+        else
+          secvalue current_state[desired.secoption.to_s]
+        end
+      end
+
+      action :set do
+        converge_if_changed :secvalue do
+          security_option = new_resource.secoption
+          security_value = new_resource.secvalue
+
+          file = Tempfile.new(["#{security_option}", ".inf"])
+          if security_option == "LockoutBadCount"
+            cmd = "net accounts /lockoutthreshold:#{security_value}"
+          elsif security_option == "ResetLockoutCount"
+            cmd = "net accounts /lockoutwindow:#{security_value}"
+          elsif security_option == "LockoutDuration"
+            cmd = "net accounts /lockoutduration:#{security_value}"
```

Comment:
Since these three values are flags passed to a windows command, they are likely not case sensitive, so you could use the mixed case version (ie, `LockoutDuration`) and avoid a spelling violation.

Additionally, you could cut down on noise a bit by turning this from if/elsif into a case statement, like:

```
          case security_option
          when "LockoutBadCount"
            cmd = "net accounts /lockoutthreshold:#{security_value}"
          when "ResetLockoutCount"
            cmd = "net accounts /lockoutwindow:#{security_value}"
          when "LockoutDuration"
            cmd = "net accounts /lockoutduration:#{security_value}"
          when "NewAdministratorName", "NewGuestName"
```

etc